src/devices/GoPiGo3/Movements/Motor.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.GoPiGo3.Models;
+using System;
+using System.Threading;
+
+namespace Iot.Device.GoPiGo3.Movements
+{
+    /// <summary>
+    /// Polarity of the motor
+    /// </summary>
+    public enum Polarity
+    {
+#pragma warning disable
+        Backward = -1, Forward = 1, OppositeDirection = 0
+#pragma warning restore
+    };
+
+    public class Motor
+    {
+        // represent the Brick
+        private GoPiGo _goPiGo = null;
+        private int _tacho;
+        private Timer _timer = null;
+        private int _periodRefresh;
+
+        /// <summary>
+        /// Create a motor
+        /// </summary>
+        /// <param name="port">Motor port</param>
+        public Motor(GoPiGo brick, MotorPort port) : this(brick, port, 1000) { }
+
+        public Motor(GoPiGo brick, MotorPort port, int timeout)
```

Comment:
Do we need to do null check or other input validation (like negative timeout)?

Reply:
negative timeout are ok, at least -1 (it's infinite). For the GoPiGo class, if null, exception will be raised at first call. I can check if you think I have too.